didcomm_resolver/resolver.py

```diff
@@ -89,32 +89,33 @@ async def update_connection(
         cls,
         session: ProfileSession,
         connection_id: str,
-        methods: Sequence[str],
+        methods: Set[str],
     ):
         """Update resolvers supported methods."""
         return await cls._upsert_methods(session, connection_id, methods, update=True)
 
     @classmethod
-    async def _upsert_methods(cls, session, connection_id, methods, update=False):
-
-        conn_record = await ConnRecord.retrieve_by_id(session, connection_id)
-        conn_record = cast(ConnRecord, conn_record)
-
-        if not update:
+    async def _upsert_methods(
+        cls, session, connection_id, methods: Set[str], update=False
+    ):
+        # retrieve connection record for metadata lookup, where methods are persisted
+        conn_record: ConnRecord = cast(
+            ConnRecord, await ConnRecord.retrieve_by_id(session, connection_id)
+        )
+        if update:
```

Comment:
Update means to replace the values of the previous methods with the newest.
If it is not an update, it just adds the newest methods to the current ones.

Reply:
O, I think I understand now.